org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/testdetection/TestFinderHandler.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.core.internal.testdetection;
+
+import static java.util.Collections.emptyList;
+import static org.eclipse.jdt.ls.core.internal.JDTUtils.resolveCompilationUnit;
+
+import java.util.List;
+
+/** Class for finding test methods in the different areas. */
+public class TestFinderHandler {
+  private static final String FILE_CONTEXT_TYPE = "FILE";
+  private static final String FOLDER_CONTEXT_TYPE = "FOLDER";
+  private static final String SET_CONTEXT_TYPE = "SET";
+  private static final String PROJECT_CONTEXT_TYPE = "PROJECT";
+  private static final String CURSOR_POSITION_CONTEXT_TYPE = "CURSOR_POSITION";
+
+  private static String fileUri;
+  private static String projectUri;
+  private static String testMethodAnnotation;
+  private static String testClassAnnotation;
+  private static String contextType;
+  private static List<String> classes;
+  private static int cursorOffset;
+
+  /**
+   * Finds tests in some area. The area is defined by first parameter of arguments - it is context
+   * type.
+   *
+   * <p>The context type can be:
+   *
+   * <ul>
+   *   <li>FILE - find tests in the file
+   *   <li>FOLDER - find tests in the folder/package
+   *   <li>SET - find tests in custom set of classes
+   *   <li>PROJECT - find tests in the project
+   *   <li>CURSOR_POSITION - find class in cursor position
+   * </ul>
+   *
+   * The values of arguments are:
+   *
+   * <p>0: context type (String) 1: project URI (String) 2: fqn of test method annotation (String)
+   * 3: fqn of test class annotation (String). If context type is CURSOR_POSITION 4: file URI
+   * (String) 5: cursor offset (double) If context type is FILE or FOLDER 4: file URI (String) If
+   * contex type is SET 4: list of classes (List<String>)
+   *
+   * @param arguments list of arguments
+   * @return test methods' declarations
+   */
+  public static List<String> findTests(List<Object> arguments) {
+    readArguments(arguments);
+    JavaTestFinder finder = new JavaTestFinder();
+
+    switch (contextType) {
```

Comment:
These really should be separate commands. Then we would not have to dispatch on parameter values.

Reply:
agree and it'll be more clearly. Will do it